### DISCUSSION TITLE
coroutine 'AsyncServer.enter_room' was never awaited

### DISCUSSION THREAD

**Discussion by zpeng2-bert** [zauberzeug/nicegui#4090] [Q&A] [answered]
### Question

I was originally developing in Python version 3.12.0 and the latest version of nicegui, but due to some reasons, I needed to downgrade the Python version to 3.7.6. However, I didn't know which version of nicegui it should correspond to, which resulted in an error message called Runtime Warning: coroutine 'AsyncServer.enter_room' was never awaited。 Which version of Nicegui should I use?
![微信图片_20241211143616](https://github.com/user-attachments/assets/934e50ac-6df7-4dae-a3a7-4cf03e07303b)

**Comment by falkoschindler** [comment, marked as answer]
Hi @zpeng2-bert,

The last version supporting Python 3.7 was NiceGUI 1.2.24, as indicated by the release notes of [1.3.0](https://github.com/zauberzeug/nicegui/releases/tag/v1.3.0).
Python 3.7 had its end-of-life in June 2023, so I wouldn't recommend using it anymore.

**Reply by zpeng2-bert** [reply]
Yeeep, I also don't want to use this version of Python, but as you know, there are always many things that cannot be determined by oneself. Thank you very much for your answer.
